packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -590,13 +597,13 @@ function propertySort(itemsKey, sortPropertiesKey) {
       return [prop, direction];
     });
 
-    // TODO: Ideally we'd only do this if things have changed
-    // Add observers
-    normalizedSort.forEach(prop => {
-      var args = [this, `${itemsKey}.@each.${prop[0]}`, didChange];
-      cp._sortPropObservers.push(args);
-      addObserver.apply(null, args);
-    });
+    sortDependentKeys = normalizedSort.map(prop => `${itemsKey}.@each.${prop[0]}`);
+
+    this[sym] = sortDependentKeys;
+
+    // NOTE: This mechanism of dynamically adding/remove DK’s is not public API.
+    // I is likely something that should eventually be explored,
+    cp._dependentKeys.push.apply(cp._dependentKeys, sortDependentKeys);
```

Comment:
I don't understand this.  Calling `.property` on a CP updates its dependent keys, this is definitely public API...